examples/Microsoft.Insights/workbooks/main.tf

```diff
@@ -0,0 +1,65 @@
+terraform {
+  required_providers {
+    azapi = {
+      source  = "azure/azapi"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+provider "azapi" {
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "myResourceGroup"
+  location = "westus"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "mystorage"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_service_plan" "test" {
+  name                = "my-app-service-plan"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  os_type             = "Linux"
+  sku_name            = "Y1"
+}
+
+resource "azurerm_linux_function_app" "test" {
+  name                = "my-linux-function-app"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  storage_account_name = azurerm_storage_account.test.name
+  service_plan_id      = azurerm_service_plan.test.id
+
+  site_config {}
+}
+
+resource "azapi_resource" "test" {
+  type      = "Microsoft.Insights/workbooks@2021-08-01"
+  name      = "deadb33f-8bee-4d3b-a059-9be8dac93960"
+  parent_id = azurerm_resource_group.test.id
+  location  = azurerm_resource_group.test.location
+  body = jsonencode({
+    kind = "user"
+    properties = {
+      category       = "workbook"
+      displayName    = "My Workbook"
+      serializedData = "{\"version\":\"Notebook/1.0\",\"items\":[{\"type\":1,\"content\":\"{\"json\":\"## New workbook\\r\\n---\\r\\n\\r\\nWelcome to your new workbook.  This area will display text formatted as markdown.\\r\\n\\r\\n\\r\\nWe've included a basic analytics query to get you started. Use the `Edit` button below each section to configure it or add more sections.\"}\",\"halfWidth\":null,\"conditionalVisibility\":null},{\"type\":3,\"content\":\"{\"version\":\"KqlItem/1.0\",\"query\":\"union withsource=TableName *\\n| summarize Count=count() by TableName\\n| render barchart\",\"showQuery\":false,\"size\":1,\"aggregation\":0,\"showAnnotations\":false}\",\"halfWidth\":null,\"conditionalVisibility\":null}],\"isLocked\":false}"
```

Comment:
Can this be heredoc for readability?

Reply:
It has another encoded json string inside, so I prefer to keep it as it is in the example.